From dbc8ab76b9f50048f8cf73a792cfded7c1061023 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Wed, 15 Jan 2014 12:19:08 +0100
Subject: [PATCH 02/10] ssm: btrfs backend: try to find the real device

Currently btrfs backend is trying to use the "device" directly from the
output. However in recent btrfs it has been changed to print out device
names in format (/dev/mapper/whatever) so we should really use
get_real_device() function to obtain the "real" device name.

Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---
 ssmlib/backends/btrfs.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssmlib/backends/btrfs.py b/ssmlib/backends/btrfs.py
index eb23403..750a517 100644
--- a/ssmlib/backends/btrfs.py
+++ b/ssmlib/backends/btrfs.py
@@ -119,7 +119,7 @@ class Btrfs(template.Backend):
                 fs_used = float(misc.get_real_size(array[6]))
 
             elif array[0] == 'devid':
-                dev['dev_name'] = array[7]
+                dev['dev_name'] = misc.get_real_device(array[7])
 
                 if not pool_name:
                     pool_name = self._find_uniq_pool_name(label, array[7])
-- 
1.8.3.1

